### PR TITLE
Feature/json serializer

### PR DESCRIPTION
**Pull request by sas2job**

### CODE REVIEW

File: app/serializers/user_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class UserSerializer
+	include JSONAPI::Serializer
+	attributes :id, :name, :email, :password, :password_confirmation
```
Comment (alxekb) [inline app/serializers/user_serializer.rb:3]: `set_key_transform :camel_lower` to match JS naming convension.

File: spec/serializers/user_serializer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe UserSerializer, type: :serializer do
+	describe 'as_json' do
+		it 'includes name and email' do
+			spec_user = FactoryBot.build( :user, name: 'specuser', email: 'specuser@email.com' )
+			serialized = JSON.parse(spec_user.serializable_hash.to_json)
+			expect( serialized[ "name" ] ).to eq "specuser"
+      expect( serialized[ "email" ] ).to eq "specuser@email.com"
```
Comment (alxekb) [inline spec/serializers/user_serializer_spec.rb]: Please fix indentaion 😃

File: app/controllers/api/v1/users_controller.rb [hunk 3]
```diff
@@ -6,13 +6,13 @@ class UsersController < ApplicationController
...
 				@users = User.all
-				render json: @users
+				render json: UserSerializer.new( @users ).serializable_hash.to_json
 			end
 
 			def show
 				ExceptionHandler.record_not_found if @user.blank?
 
-				render json: @user
+				render json: UserSerializer.new( @user ).serializable_hash.to_json
```
Comment (alxekb) [inline app/controllers/api/v1/users_controller.rb]: Please extract to a private method

File: app/controllers/api/v1/users_controller.rb [hunk 4]
```diff
@@ -62,7 +62,7 @@ def user_params
 			end
 
 			def set_user
-				@user = User.find( id: params[ :id ] )
+				@user = User.find( params[ :id ] )
```
Comment (alxekb) [inline app/controllers/api/v1/users_controller.rb:62]: I can't see any Exception handler here.

**Review by alxekb — CHANGES_REQUESTED** [review]
🔥

File: spec/serializers/user_serializer_spec.rb (hunk 2 above)
Comment (sas2job) [reply spec/serializers/user_serializer_spec.rb]: > Please fix indentaion

Large indents create .rubocop.yml

File: app/controllers/api/v1/users_controller.rb (hunk 3 above)
Comment (sas2job) [reply app/controllers/api/v1/users_controller.rb]: > Please extract to a private method

Done

File: app/controllers/api/v1/users_controller.rb (hunk 4 above)
Comment (sas2job) [reply app/controllers/api/v1/users_controller.rb:62]: > I can't see any Exception handler here.

Done.

File: app/controllers/api/v1/users_controller.rb [hunk 5]
```diff
@@ -6,13 +6,14 @@ class UsersController < ApplicationController
 			before_action :set_user, only: %i[show update destroy]
 			def index
 				@users = User.all
-				render json: @users
+
+				render_json(@users)
 			end
 
 			def show
 				ExceptionHandler.record_not_found if @user.blank?
```
Comment (alxekb) [inline app/controllers/api/v1/users_controller.rb:13]: Please remove this line

File: app/controllers/api/v1/users_controller.rb [hunk 6]
```diff
@@ -62,7 +63,12 @@ def user_params
 			end
 
 			def set_user
-				@user = User.find( id: params[ :id ] )
+				@user = User.find( params[ :id ] )
+				ExceptionHandler.record_not_found if @user.blank?
+			end
+
+			def render_json(user)
```
Comment (alxekb) [inline app/controllers/api/v1/users_controller.rb]: Move this method to a module and include module at the Application controller.

File: app/controllers/api/v1/users_controller.rb [hunk 7]
```diff
@@ -62,7 +63,12 @@ def user_params
 			end
 
 			def set_user
-				@user = User.find( id: params[ :id ] )
+				@user = User.find( params[ :id ] )
+				ExceptionHandler.record_not_found if @user.blank?
```
Comment (alxekb) [inline app/controllers/api/v1/users_controller.rb]: If you chose #find it's better to catch RecoedNotFound exception at the controller method.

**Review by alxekb — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/users_controller.rb (hunk 5 above)
Comment (sas2job) [reply app/controllers/api/v1/users_controller.rb:13]: Done

File: app/controllers/api/v1/users_controller.rb (hunk 6 above)
Comment (sas2job) [reply app/controllers/api/v1/users_controller.rb]: > Move this method to a module and include module at the Application controller.

Done.
https://github.com/sas2job/test-rails-api/blob/feature/json_serializer/app/controllers/concerns/exception_handler.rb

**Review by alxekb — APPROVED** [review]

**Pull request closed without merging**